### PR TITLE
Tweak `Lint/UnneededDisable` cop behavior.

### PR DESCRIPTION
**Pull request by tdeo**
I was noticing several weird behaviours with UnneededDisable.

First one is linked to https://github.com/bbatsov/rubocop/issues/4459,
it turns out the following piece of code:
```
  # rubocop:disable Lint/UnneededDisable
  # rubocop:disable all
```
returns the offense "Unnecessary disabling of all cops." where I believe
it shouldn't. It seems this was introduced by @rrosenblum changes
in `UnneededDisable#check` in https://github.com/bbatsov/rubocop/commit/d042ccff

Second point I noticed and wanted to address is that the following code:
```
  # rubocop:disable Lint/UnneededDisable
  # rubocop:disable Bundler/OrderedGems
  # rubocop:disable Bundler/OrderedGems
```
produces an offense for unnecessary disabling on line 3. This shouldn't be
reported as the cop is disabled. This commit addresses it as well.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests(`rake spec`) are passing.
* [x] The new code doesn't generate RuboCop offenses that are checked by `rake internal_investigation`.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by rrosenblum** [comment]
The commits will need to be squashed. 

Linking for context, the commit that you mentioned was introduced in #1967 as a fix for #1959.

File: spec/rubocop/cop/lint/unneeded_disable_spec.rb [hunk 1]
```diff
@@ -273,6 +305,25 @@
...
+                 '# rubocop:disable all'].join("\n")
+              end
+
+              let(:cop_disabled_line_ranges) do
+                { 'Lint/UnneededDisable' => [1..Float::INFINITY],
+                  'all' => [2..Float::INFINITY] }
+              end
+
+              it 'does not return an offense' do
+                puts cop.offenses
```
Comment (rrosenblum) [inline spec/rubocop/cop/lint/unneeded_disable_spec.rb]: This looks like it may have been left in by mistake.

**Comment by bbatsov** [comment]
This issue reminded me that we should eventually name the cop `UnneedCopDisableDirective` or something else more informative.

**Comment by schneems** [comment]
I’m getting this error with rubocop 0.55.0 did this feature ship?


lib/tasks/heroku_clear_tasks.rake:1:1: W: Lint/UnneededCopDisableDirective: Unnecessary disabling of Lint/UnneededDisable (unknown cop).
# rubocop:disable Lint/UnneededDisable

**Comment by tdeo** [comment]
@schneems Probably because of this commit released in 0.53.0: https://github.com/bbatsov/rubocop/commit/a0870fb747f95a85349d91acff4f6030c257a663#diff-b1f19ab29b0a1eac6d4690192e23c42a

Changing your statement to `# rubocop:disable Lint/UnneededDisable, Lint/UnneededCopDisableDirective` should hopefully be both pre-0.53 and post-0.53 compatible

**Pull request merged**